app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,286 @@ def registry_authorize
+      if @product.nil?
+        return render_podman_error(
+          "NAME_UNKNOWN",
+          "Product id not found: '#{prod_id}'",
+          :not_found
+        )
+      end
+      true
+    end
+
+    def get_matching_products_from_org(organization, product_label)
+      return organization.products.where("LOWER(label) = '#{product_label}'") # convert to lowercase
+    end
+
+    def get_root_repo_from_product(product, root_repo_name)
+      return product.root_repositories.where(label: root_repo_name)
+    end
+
+    def check_blob_push_container(props)
+      unless props[:name].present? && props[:name].length > 0
+        return render_podman_error(
+          "NAME_INVALID",
+          "Invalid format. Container name cannot be blank.",
+          :bad_request
+        )
+      end
+
+      @container_name = props[:name]
+      @container_path_input = "#{props[:organization]}/#{props[:product]}/#{props[:name]}"
+      @container_push_name_format = props[:schema]
+
+      # If the repo already exists, check if the existing push format matches 
+      root_repo = get_root_repo_from_product(@product, props[:name]).first
+      if !root_repo.nil? && @container_push_name_format != root_repo.container_push_name_format
+        return render_podman_error(
+          "NAME_INVALID",
+          "Repository name '#{@container_name}' already exists in this product using a different naming scheme. Please retry your request with the #{root_repo.container_push_name_format} format or destroy and recreate the repository using your preferred schema.",
+          :conflict
+        )
+      end
+
+      true
+    end
+
+    def create_container_repo_if_needed
+      if get_root_repo_from_product(@product, @container_name).empty?
+        root = @product.add_repo(
+          name: @container_name,
+          label: @container_name,
+          download_policy: 'immediate',
+          content_type: Repository::DOCKER_TYPE,
+          unprotected: true,
+          is_container_push: true,
+          container_push_name: @container_path_input,
+          container_push_name_format: @container_push_name_format
+        )
+        sync_task(::Actions::Katello::Repository::CreateRoot, root, @container_path_input)
+      end
+    end
+
+    def blob_push_cleanup
+      # after manifest upload, index content and set version href using pulp api
+      root_repo = @product.root_repositories.where(label: @container_name)&.first
+      instance_repo = root_repo&.library_instance
+
+      unless root_repo.present? && instance_repo.present?
+        return render_podman_error(
+          "BLOB_UPLOAD_UNKNOWN",
+          "Could not locate local uploaded repository for content indexing.",
+          :not_found
+        )
+      end
+
+      api = ::Katello::Pulp3::Repository.api(SmartProxy.pulp_primary, ::Katello::Repository::DOCKER_TYPE).container_push_api
+      api_response = api.list(name: @container_path_input).results&.first
```

Comment:
For ID-type repositories, this always returns empty because @container_path_input does not include "id":

```
[18] pry(#<Katello::Api::Registry::RegistryProxiesController>)> @container_path_input
=> "6/344/arianna"

------------------------------------

[19] pry(#<Katello::Api::Registry::RegistryProxiesController>)> api.list(name: @container_path_input)
2024-06-03T19:05:41 [D|kat|334ab382] Calling API: RepositoriesContainerPushApi.list ...
2024-06-03T19:05:41 [D|kat|334ab382] HTTP response body ~BEGIN~
 334ab382 | {"count":0,"next":null,"previous":null,"results":[]}
 334ab382 | ~END~
 334ab382 | 
2024-06-03T19:05:41 [D|kat|334ab382] API called: RepositoriesContainerPushApi#list
 334ab382 | Data: #<PulpContainerClient::PaginatedcontainerContainerPushRepositoryResponseList:0x00007fba5ca9f358 @count=0, @results=[]>
 334ab382 | Status code: 200
 334ab382 | Headers: {"date"=>"Mon, 03 Jun 2024 19:05:41 GMT", "server"=>"gunicorn", "content-type"=>"application/json", "vary"=>"Accept,Cookie", "allow"=>"GET, HEAD, OPTIONS", "x-frame-options"=>"DENY", "content-length"=>"52", "x-content-type-options"=>"nosniff", "referrer-policy"=>"same-origin", "cross-origin-opener-policy"=>"same-origin", "correlation-id"=>"334ab382-a357-4489-98a1-e909c19b1d41", "access-control-expose-headers"=>"Correlation-ID", "via"=>"1.1 centos9-katello-devel.manicotto.example.com"}
=> #<PulpContainerClient::PaginatedcontainerContainerPushRepositoryResponseList:0x00007fba5ca9f358 @count=0, @results=[]>

-----------------------------------------

  {
    "pulp_created": "2024-06-03T19:01:47.382749Z",
    "latest_version_href": "/pulp/api/v3/repositories/container/container-push/018fdf7a-a2f5-7f56-bb36-2e9b66fd0f19/versions/1/",
    "pulp_labels": {},
    "retain_repo_versions": null,
    "pulp_href": "/pulp/api/v3/repositories/container/container-push/018fdf7a-a2f5-7f56-bb36-2e9b66fd0f19/",
    "pulp_last_updated": "2024-06-03T19:01:48.179714Z",
    "versions_href": "/pulp/api/v3/repositories/container/container-push/018fdf7a-a2f5-7f56-bb36-2e9b66fd0f19/versions/",
    "name": "id/6/344/arianna",
    "manifest_signing_service": null,
    "description": null
  },

```

Reply:
I believe this is the cause for ID-type repositories not being created correctly.

Reply:
This is fixed as of the next update to this PR, thank you